utils: fixed shell vulnerability in mkrest.py

jadenabrams100 opened the pull request:
I found a vulnerability in this mkrest.py utilities file, where malformed input in the src_file variable could give an attacker a shell if they escaped the rest of the command. This fix restructures the Popen call to not use shell=True while still retaining the same functionality.

echoix commented on utils/mkrest.py:
```diff
@@ -81,8 +81,9 @@ def read_file(name):
     sys.stdout.write(tmp_data)
 
+arguments = ['pandoc', '-s', '-r', 'html', src_file, '-w', 'rst']
 process = subprocess.Popen(
-    "pandoc -s -r html %s -w rst" % src_file, shell=True, stdout=subprocess.PIPE
+    arguments, stdout=subprocess.PIPE
```
As I reviewed in https://github.com/OSGeo/grass/pull/3436#discussion_r1493954603, it was all good for me!

echoix commented:
Thanks for your contribution and taking the time to make a clean PR for this!

echoix commented:
It will just need to run black to fix the formatting error as shown in the linter logs. In case you already ran black, note that we haven't updated yet to the 2024 style, just by lack of bandwidth to update the code base beforehand.

jadenabrams100 commented:
> It will just need to run black to fix the formatting error as shown in the linter logs. In case you already ran black, note that we haven't updated yet to the 2024 style, just by lack of bandwidth to update the code base beforehand.

My apologies, I not have pre-commit set up when I made my changes and linting escaped my mind. I have fixed the formatting and added the commit to this PR!
